### ISSUE TITLE
country_crossing_penalty 

### ISSUE THREAD

**Issue by iddqd2d** [valhalla/valhalla#4126]
Hi. I set the country_crossing_penalty to a very high value to avoid country crossings, but it didn't work.

From {lat: 42.905086, lon: -85.53690} , to: {lat: 43.005185, lon: -78.407109}

![image](https://github.com/valhalla/valhalla/assets/40464324/0a1bab5c-6953-429c-885f-53eadedbef6f)

**Comment by iddqd2d** [comment]
Yes. Use:  "valhalla_build_admins --config ./conf/valhalla.json north-america-latest.osm.pbf"
It works with other coordinates:

costing_options:{"auto":{country_crossing_penalty:0, country_crossing_cost: 0}}:
![image](https://github.com/valhalla/valhalla/assets/40464324/5aced5e7-d5ef-43a6-ac03-a2d2f377b701)


costing_options:{"auto":{country_crossing_penalty:20000.0, country_crossing_cost: 0}}
![image](https://github.com/valhalla/valhalla/assets/40464324/e7f7cd95-9049-4425-ad5e-e14a2cf6fa56)

**Comment by iddqd2d** [comment]
Can you check on your tiles? Please
From {lat: 42.905086, lon: -85.53690} , to: {lat: 43.005185, lon: -78.407109}

**Comment by nilsnolde** [comment]
I guess you checked the public server? (on a phone right now)

That’s „my“ tiles with a fairly recent Valhalla version.

**Comment by iddqd2d** [comment]
URL: https://valhalla.openstreetmap.de/directions?profile=bus&wps=-83.73229574747475%2C42.99655818181818%2C-77.662816%2C43.1492614


![image](https://github.com/valhalla/valhalla/assets/40464324/e2f086ce-6a14-4160-9499-b912ec23753d)

**Comment by iddqd2d** [comment]
Sorry but how can i get route without border crossing?

**Comment by kevinkreiser** [comment]
my thoughts exactly. we will need to mark this as troubleshooting and do some actual debugging

**Comment by kevinkreiser** [comment]
i just had a look at the code and i think this is probably a case of the data not being tagged that there is a border control point in certain places. it seems the algorithm only penalizes in the case the osm node is tagged a certain way, see here: https://github.com/valhalla/valhalla/issues/4161#issuecomment-1631765939

**Comment by Sasha-Bachynskyi** [comment]
I've encountered the same issue and wonder if it will be solved?

**Comment by nilsnolde** [comment]
It will be at some point :) It's all about prioritization I guess (both maintainers or external contributors). Can't put a date on a fix from my side, it has entered the queue at least:)

**Comment by xxhh1112** [comment]
w

**Comment by iddqd2d** [comment]
Hello! I need to get directions without crossing a border if all my locations are in the same country. Any news about this problem?

If I use north-america-latest.osm.pbf I have the same problem.

I tried creating tiles for michigan-latest.osm.pbf, ohio-latest.osm.pbf, pennsylvania-latest.osm.pbf, new-york-latest.osm.pbf, ontario-latest.osm.pbf and got the expected result .

![image](https://github.com/valhalla/valhalla/assets/40464324/01354f44-7698-4e78-95aa-4152add77531)


But I need use north-america-latest.osm.pbf. Can I rewrite north-america-latest.osm.pbf tiles from michigan-latest.osm.pbf, ohio-latest.osm.pbf, pennsylvania-latest.osm.pbf, new-york-latest.osm.pbf, ontario-latest.osm.pbf tiles. I tried creating tiles for all the individual regions of North America, but it takes a long time and I don't know if it will work?

**Comment by nilsnolde** [comment]
As Kevin pointed out the data isn’t properly tagged. So best you can do is go into OSM and tag it right, ie set “border_control = true” on an existing/new node where it’s needed. The name of the costing option is just misleading but it does make sense if you think about it. You only want that penalty wherever there actually is a controlled border (eg not inside EU).
